### PR TITLE
Generate Dash License Report

### PR DESCRIPTION
**Pull request by wba2hi**
Closes: #2

### CODE REVIEW

File: .github/actions/check-dash/action.yml
Comment (wba2hi) [inline .github/actions/check-dash/action.yml:1]: Taken from here: https://github.com/eclipse/kuksa.val/blob/master/.github/actions/check-dash/action.yml

File: build.gradle.kts [hunk 1]
```diff
@@ -22,3 +44,40 @@ subprojects {
         systemProperties = systemPropertiesMap
     }
 }
+
+tasks.register("mergeDashFiles") {
```
Comment (wba2hi) [inline build.gradle.kts:48]: I would have prefered to put this into "dash.gradle.kts" to make it "re-usable" but unfortunately this task is only needed by the root project,... maybe it makes sense to think about a convention like naming this like dash.root.gradle.kts or so

File: buildscripts/dash.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# the following adaptions were done:
+# - filter entries marked with (n) = not resolvable
+# - filter entries marked FAILED
+# - filter entries referencing a (sub-)project
+# - change normalization step to be compatible with jetpack compose (androidx.compose.ui:ui-test-manifest -> 1.5.0)
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Linux*)     GREP="grep";; # Linux
+    Darwin*)    GREP="ggrep";; # Mac
```
Comment (wba2hi) [inline buildscripts/dash.sh:45]: To test this on Mac ggrep needs to be installed

File: buildscripts/dash.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+echo "${GREP}"
+
+./gradlew "$projectName":dependencies \
+| ${GREP} -Poh "(?<=\-\-\- ).*" \
+| ${GREP} -Pv "\([nc\*]\)" \
+| ${GREP} -Pv "FAILED" \
+| ${GREP} -Pv "project :[a-zA-Z0-9]+" \
+| perl -pe 's/([\w\.\-]+):([\w\.\-]+):(?:[\w\.\-]+ -> )?([\w\.\-]+).*$/$1:$2:$3/gmi;t' \
+| perl -pe 's/([\w\.\-]+):([\w\.\-]+) -> ([\w\.\-]+).*$/$1:$2:$3/gmi;t' \
+| sort -u \
```
Comment (wba2hi) [inline buildscripts/dash.sh:57]: there is most probably a lot of potential to "optimize"/"shrink" these terms / number of lines. I tried some stuff, but always lead to something breaking. Reducing number of lines obviously also means increasing the complexity to understand each terms, so I left it for now.

**Comment by wba2hi** [comment]
Build: https://github.com/eclipse-kuksa/kuksa-android-sdk/actions/runs/6146232536?pr=3
Dash Data: https://github.com/eclipse-kuksa/kuksa-android-sdk/suites/16034618362/artifacts/915220431

File: buildscripts/dash.sh (hunk 3 above)
Comment (Chrylo) [reply buildscripts/dash.sh:57]: Fine for me for now :)

**Comment by wba2hi** [comment]
@Chrylo Ready for Review

**Comment by SebastianSchildt** [comment]
Not sure the Dash data is "correct", at least there are a LOT of things where sources could not be found (i'd assign you in IP tracker, but so far your user seems not to be visible there)

As an example check this failed ticket: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/10457

vs. this working older one https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/10217

Is this another "Kind" of dependency? or is our dash input not optimal?

So far there seems to be around 20 issues where sources are missing https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/?sort=created_date&state=opened&author_username=sschildt&label_name%5B%5D=Help%20Wanted&first_page_size=100 
, might become more, once Dash comes aorund to scan the other issues

(Check all issues opened in my name: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/?sort=created_date&state=opened&author_username=sschildt&first_page_size=100

**Comment by SebastianSchildt** [comment]
Seems to be getting better now, wayne worked his magic and teached dash bot new tricks: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/10466  let's wait if it can churn through

**Comment by wba2hi** [comment]
> Not sure the Dash data is "correct", at least there are a LOT of things where sources could not be found (i'd assign you in IP tracker, but so far your user seems not to be visible there)
It's just as Wayne stated. Some of the android sources are only available on the google maven repository. However for older versions of libraries google didn't upload sources at all. So there might still be some libraries and versions where the sources simply can not be found except maybe by scramming through the aosp sources directly. 

I think some of the libraries have a "Android Software Development Kit" License (https://developer.android.com/studio/terms.html) which allows the use of said libraries, but is not open source (maps and navigation and stuff where google basically generates money with). 

> Is this another "Kind" of dependency? or is our dash input not optimal?
Unfortunately no and (most probably) no. Can't / won't guarantee 100% perfectness as I'm relying on code snippets here where I had to make adaptions to, but I checked them by sample and the results look "fine".  
 
> So far there seems to be around 20 issues where sources are missing https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/?sort=created_date&state=opened&author_username=sschildt&label_name%5B%5D=Help%20Wanted&first_page_size=100 , might become more, once Dash comes aorund to scan the other issues
Either this filter does not work, or they are all identified by now? The filter produces no results for me

**Comment by SebastianSchildt** [comment]
I feel - except following up on how the license clear in Eclipses backend -  from a technical perspective this PR works

@Chrylo pls review and once happy give a positive review/ping me

File: build.gradle.kts (hunk 1 above)
Comment (Chrylo) [reply build.gradle.kts:48]: I agree as soon as it makes sense to extract this one. Currently it is still pretty clean.

[the remark this answers is not in the document]
Comment (Chrylo) [reply build.gradle.kts:48]: Seems like this could be a good code comment. Maybe add something like this in the next PR.

**Comment by Chrylo** [comment]
@SebastianSchildt LGTM ;)

**Pull request merged**